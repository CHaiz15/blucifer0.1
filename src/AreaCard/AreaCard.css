.area-card {
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid white;
  border-radius: 10px;
  width: 39%;
  min-height: 210px;
  margin: 3% 3% 0 3%;
  cursor: pointer;
  background: linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5) );
  color: #fff;
}

.area-card:hover {
  animation: shadow-inset-center 0.25s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}

.card-title {
  font-size: 2em;
  text-shadow: 0 0 2px #000;
}

.card-div {
  width: 100%;
  height: 100%;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
}

.hovered-card {
  text-align: center;
  padding: 5%;
}

@keyframes shadow-inset-center {
  0% {
    box-shadow: inset 0 0 0 0 rgba(0, 0, 0, 0);
  }
  100% {
    box-shadow: inset 0 0 20px 0px rgba(0, 0, 0, 0.5);
  }
}
