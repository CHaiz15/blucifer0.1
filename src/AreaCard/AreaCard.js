import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './AreaCard.css';

  class AreaCard extends Component {
  constructor() {
    super();
    this.state = {
      hover: false
    }
  }
  hoverTrue = () => {
    this.setState({hover: true})
  }
  hoverFalse = () => {
    this.setState({hover: false})
  }

  selectArea = () => {
    this.props.addSelectedArea(this.props.id)
  }

  render() {
    let hoveredCard =
      <div value={this.props.id} onClick={this.selectArea} className='hovered-card'>
        <h2 className='short-name'>{this.props.shortName}</h2>
        <p>{this.props.about}</p>
      </div>;
    let originalCard = <h1 className="card-title">{this.props.areaName}</h1>;

    return (
      <div
        onMouseOver={this.hoverTrue}
        onMouseLeave={this.hoverFalse}
        className="card-div"
      >
        {this.state.hover ? hoveredCard : originalCard}
      </div>
    )
  }
}

export default AreaCard;

AreaCard.propTypes = {
  addSelectedArea: PropTypes.func,
  id: PropTypes.number,
  areaName: PropTypes.string,
  shortName: PropTypes.string, 
  about: PropTypes.string,
  listings: PropTypes.array,
}