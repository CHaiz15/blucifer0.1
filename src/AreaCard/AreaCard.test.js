import React from 'react';
import { render } from '@testing-library/react';
import AreaCard from './AreaCard';
import { shallow } from 'enzyme'

describe('AreaCard', () => {
  it('should be an instance of AreaCard component', () => {
    const wrapper = shallow(<AreaCard/>);
      expect(wrapper).toMatchSnapshot();
  });
})