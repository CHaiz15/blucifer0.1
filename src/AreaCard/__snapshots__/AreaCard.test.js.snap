// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaCard should be an instance of AreaCard component 1`] = `
<div
  className="card-div"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <h1
    className="card-title"
  />
</div>
`;
